internal/support/scheduler/schedule.go

```diff
@@ -0,0 +1,482 @@
+//
+// Copyright (c) 2018 Tencent
+//
+// Copyright (c) 2017 Dell Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+package scheduler
+
+import (
+	"errors"
+	"fmt"
+	"github.com/edgexfoundry/edgex-go/pkg/models"
+	queueV1 "gopkg.in/eapache/queue.v1"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	ScheduleInterval                  = 500
+)
+
+//the schedule specific shared variables
+var (
+	mutex                          sync.Mutex
+	scheduleQueue                  = queueV1.New()                     // global schedule queue
+	scheduleIdToContextMap         = make(map[string]*ScheduleContext) // map : schedule id -> schedule context
+	scheduleNameToContextMap       = make(map[string]*ScheduleContext) // map : schedule name -> schedule context
+	scheduleEventIdToScheduleIdMap = make(map[string]string)           // map : schedule event id -> schedule id
+)
+
+func StartTicker() {
+	go func() {
+		for range ticker.C {
+			triggerSchedule()
+		}
+	}()
+}
+
+func StopTicker() {
+	ticker.Stop()
+}
```

Comment:
This function is unused. Should it be called from init.go::Destruct()?

Reply:
reserved for unit tests which will be added. If there is a better paradigm for adding helper functions for unit tests I would love to do it.